src/bindings/python/flux/job.py

```diff
@@ -267,6 +267,155 @@ def wait(flux_handle, jobid=lib.FLUX_JOBID_ANY):
+        return result
+
+    def reset(self):
+        super().reset()
+        if self.results:
+            self.results.popleft()
+
+
+def event_watch_async(flux_handle, jobid, name="*", eventlog="eventlog"):
+    """Asynchronously get eventlog updates for a job
+
+    Asynchronously watch the events of a job eventlog, optionally only
+    returning events that match a glob pattern.
+
+    Returns a JobEventWatchFuture. Call .get_event() from the then
+    callback to get the currently returned event from the Future object.
+
+    The Future must be reset with future.reset() before leaving the
+    continuation callback to reset the current result and queue the next
+    result event.
+
+    :param flux_handle: handle for Flux broker from flux.Flux()
+    :type flux_handle: Flux
+    :param jobid: the job ID on which to watch events
+    :param name: The event name or glob pattern for which to wait (default: *)
+    :param eventlog: eventlog path in job kvs directory (default: eventlog)
+    :returns: a JobEventWatchFuture object
+    :rtype: JobEventWatchFuture
+    """
+    from fnmatch import fnmatch
+
+    flags = 0
+    wait_future = JobEventWatchFuture(flux_handle)
+
+    def event_watch_cb(future, arg):
+        nonlocal wait_future
+        nonlocal name
+        result = ffi.new("char *[1]")
+        try:
+            RAW.event_watch_get(future, result)
+        except OSError as exc:
+            if exc.errno == errno.ENODATA:
+                wait_future.fulfill(None)
+            else:
+                wait_future.fulfill_error(exc.errno, future.error_string())
+            return
+        event = EventLogEvent(ffi.string(result[0]).decode("utf-8"))
+        if fnmatch(event.name, name):
+            wait_future.fulfill(event)
+        future.reset()
+
+    Future(RAW.event_watch(flux_handle, int(jobid), eventlog, flags)).then(
+        event_watch_cb
+    )
+
+    return wait_future
+
+
+def event_wait(flux_handle, jobid, name, eventlog="eventlog"):
+    """Wait for a job eventlog entry 'name'
+
+    Wait synchronously for an eventlog entry named "name" and
+    return the entry to caller, or None if event never happened.
+
+    :param flux_handle: handle for Flux broker from flux.Flux()
+    :type flux_handle: Flux
+    :param jobid: the job ID on which to wait for eventlog events
+    :param name: The event name for which to wait
+    :param eventlog: eventlog path in job kvs directory (default: eventlog)
+    :returns: an EventLogEntry object, or None if no more eventrs
+    :rtype: EventLogEntry
+
+    """
+
+    flags = 0
+    result = ffi.new("char *[1]")
+
+    future = Future(RAW.event_watch(flux_handle, int(jobid), eventlog, flags))
+    while True:
+        RAW.event_watch_get(future, result)
+        if result == ffi.NULL:
+            return None
+        event = EventLogEvent(json.loads(ffi.string(result[0]).decode("utf-8")))
+        if event.name == name:
+            return event
+        future.reset()
```

Comment:
Could this take advantage of `event_watch_async` to reduce code duplication?

```python
return job.event_watch_async(flux_handle, jobid, name=name, eventlog=eventlog).get()
```

EDIT: I initially proposed the following loop until I realized that `event_watch_async` supports the name argument.

```python
for event in job.event_watch(flux_handle, jobid):
    if event.name == name:
        return event
return None
```

Reply:
Unfortunately, no. See the above comment. If `event_watch_async().get_event()` is called synchronously it returns `-1`.